koku/api/settings/settings.py

```diff
@@ -242,7 +242,13 @@ def _tag_key_handler(self, settings):
                 raise ValidationError(error_obj(key, message))
 
             if "aws" in provider_name:
-                updated[ix] = update_enabled_keys(self.schema, enabled_tag_keys, enabled_tags_no_abbr)
+                existing_enabled_tags = []
+                for key in enabled_tag_keys.objects.all():
+                    if key.enabled:
+                        existing_enabled_tags.append(key.key)
```

Comment:
I worry about the performance of this if there are a lot of tags to deal with. We should be able to lean on Django more here to filter on the enabled tags instead of looping thru them.

This query should work here. It also makes the DB do all the work instead of iterating thru the list in Python.
```suggestion
                existing_enabled_tags = list(enabled_tag_keys.objects.filter(enabled=True).values_list('key', flat=True))

```